user-access-handlers/src/main/java/no/unit/nva/handlers/IdentityServiceInitHandler.java

```diff
@@ -45,33 +32,36 @@ public class IdentityServiceInitHandler extends ApiGatewayHandler<Void, RoleList
     
     private final IdentityService identityService;
     private final CustomerService customerService;
+    private final RoleSource roleSource;
     
     @JacocoGenerated
     public IdentityServiceInitHandler() {
-        this(IdentityService.defaultIdentityService(), defaultCustomerService());
+        this(IdentityService.defaultIdentityService(), defaultCustomerService(), new DefaultRoleSource());
     }
     
-    public IdentityServiceInitHandler(IdentityService identityService, CustomerService customerService) {
+    public IdentityServiceInitHandler(IdentityService identityService, CustomerService customerService,
+                                      RoleSource roleSource) {
         super(Void.class);
         this.identityService = identityService;
         this.customerService = customerService;
+        this.roleSource = roleSource;
     }
     
     @Override
     protected RoleList processInput(Void input, RequestInfo requestInfo, Context context) {
-        var defaultRoles = createDefaultRoles()
+        var defaultRoles = roleSource.roles()
             .stream()
-            .map(attempt(this::addRole))
+            .map(attempt(this::addOrUpdateRole))
```

Comment:
Had an issue just now myself that error handling were not working correctly when using lambda annotation instead of regular method call in an attempt call. Not sure why, but have you tested what happens if this produce an exception?